lab_2_keywords_cooccurrence/main.py

```diff
@@ -29,7 +69,30 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not(check_sequence(phrases, str, False) and not isinstance(phrases, str)
+           and check_sequence(stop_words, str, False) and not isinstance(stop_words, str)):
+        return None
+
+    splt_phrases = [phrase.lower().split() for phrase in phrases]
+    candidate_keyword_phrases = []
+    keyword_phrase = []
+
+    for splt_phrase in splt_phrases:
+        for idx, word in enumerate(splt_phrase):
+
+            if word not in stop_words:
+                if idx == 0 and keyword_phrase:
+                    candidate_keyword_phrases.append(tuple(keyword_phrase))
+                    keyword_phrase = []
+                keyword_phrase.append(word)
+                if idx == (len(splt_phrase) - 1) and splt_phrases.index(splt_phrase) == len(splt_phrases) - 1:
```

Comment:
method `index` hides the whole cycle over all the elements of the sequence, so its not exaclty efficient
it is better to directly access the last element in the sequence using [-1] for example

furthermore, `index` returns the index of the _first_ occurence of the element in the sequence, so in case there are duplicate phrases it will not work correctly

Reply:
i replaced it with 
`if splt_phrase[-1] == word and splt_phrases[-1] == splt_phrase:`
wouldn't it be a problem if a word isn't the last element but just its duplicate though?

Reply:
yeah, it will be the problem in case if there are duplicate phrases

to avoid it, consider moving this block of code out of the cycle: if cycle over words has ended and there is still something in `keyword_phrase`, then it should be added to the candidate phrases